nova/virt/vmwareapi/vmops.py

```diff
@@ -1491,25 +1491,20 @@ def _destroy_instance(self, context, instance, destroy_disks=True):
         # Destroy a VM instance
         try:
             vm_ref = vm_util.get_vm_ref(self._session, instance)
-
+            vm_groups = []
             server_group_info = vm_util._get_server_group(context, instance)
-            if server_group_info:
-                cluster = cluster_util.fetch_cluster_properties(self._session,
-                                                                       vm_ref)
-                config_info_ex = cluster.propSet[0].val
+            cluster = cluster_util.fetch_cluster_properties(self._session,
+                                                            vm_ref)
+            config_info_ex = cluster.propSet[0].val
 
-                if hasattr(config_info_ex, 'group'):
-                    for key, group in enumerate(config_info_ex.group):
-                        if not hasattr(group, 'vm'):
-                            continue
-
-                        for vm in group.vm:
-                            if vm.value == vm_ref.value and len(group.vm) == 1:
-                                cluster_util.delete_vm_group(
-                                            self._session, cluster.obj,
-                                            config_info_ex.group[key])
-                                break
-                        break
+            if hasattr(config_info_ex, 'group'):
+                for group in config_info_ex.group:
+                    if not hasattr(group, 'vm'):
+                        continue
+                    for vm in group.vm:
+                        if vm.value == vm_ref.value:
+                            vm_groups.append(group.name)
+                            break
```

Comment:
I still think you should only add a group to your list if it's an openstack-created group and not an admin-created group, i.e. if it's in `server_group_info`. I don't want you to delete my manually-assigned groups, I as a cluster operator need for maintenance on the cluster or whatever.